### PR TITLE
feat: add support for edge transformation

### PR DESCRIPTION
**Pull request by koladilip**
# Description

This is done for POC and we will be making further changes.

## Linear Ticket

https://linear.app/rudderstack/issue/INT-1928/poc-for-edge-transformation
Resolves INT-1928

## Security

- [ ] The code changed/added as part of this pull request won't create any security issues with how the software is being used.

### CODE REVIEW

File: router/transformer/transformer.go [hunk 1]
```diff
@@ -597,6 +600,10 @@ func getBatchURL() string {
 	return strings.TrimSuffix(config.GetString("DEST_TRANSFORM_URL", "http://localhost:9090"), "/") + "/batch"
 }
 
+func getEdgeTransformURL() string {
+	return strings.TrimSuffix(config.GetString("DEST_TRANSFORM_URL", "http://localhost:9090"), "/") + "/edgeTransform"
```
Comment (lvrach) [inline router/transformer/transformer.go:604]: suggestion: url.[JoinPath](https://cs.opensource.google/go/go/+/go1.22.2:src/net/url/url.go;l=1265)

```suggestion
	return url.JoinPath(config.GetString("DEST_TRANSFORM_URL", "http://localhost:9090"), "/edgeTransform")
```

File: router/transformer/transformer.go [hunk 2]
```diff
@@ -148,6 +149,8 @@ func (trans *handle) Transform(transformType string, transformMessage *types.Tra
 		url = getBatchURL()
 	} else if transformType == ROUTER_TRANSFORM {
 		url = getRouterTransformURL()
+	} else if transformType == EDGE_TRANSFORM {
```
Comment (lvrach) [inline router/transformer/transformer.go:152]:
```go

switch transformType {
case BATCH:
    url = getBatchURL()
case ROUTER_TRANSFORM: 
    url = getRouterTransformURL()
case EDGE_TRANSFORM:
    url = getEdgeTransformURL()
}

**Review by lvrach — COMMENTED** [review]
some minor readability comments

File: router/transformer/transformer.go (hunk 1 above)
Comment (koladilip) [reply router/transformer/transformer.go]: addressed

File: router/transformer/transformer.go (hunk 2 above)
Comment (koladilip) [reply router/transformer/transformer.go]: addressed

**Pull request closed without merging**